src/AbstractComponent.php

```diff
@@ -78,6 +84,6 @@ public function __toString()
      */
     public function sameValueAs(ComponentInterface $component)
     {
-        return $this->__toString() == $component->__toString();
```

Comment:
the "===" is not needed as we know that we are comparing string so it won't add or make any difference there